addons/website/static/src/js/menu/content.js

```diff
@@ -703,6 +703,9 @@ var EditMenuDialog = weWidgets.Dialog.extend({
             menuType: menuType,
         });
         dialog.on('save', this, link => {
+            if (link.url.startsWith('#')) {
```

Comment:
```suggestion
            if (link.url.startsWith('#') and link.url !== '#') {
```
or
```suggestion
            if (link.url.startsWith('#') and link.length > 1) {
```
Otherwise it will also prevent to create "container" menu, see https://github.com/odoo-dev/odoo/commit/b65cb86697debe1ad3043483d3aafac40dd9c9a4.
At the end, you can probably just do it in python (in the endpoint rather than in the multiple caller) 
```py
if '#' in menu['url']:
    # Multiple case possible:
    # 1. Just "#" for (dropdown) container menu
    # 2. "#something" for anchor on current page
    # 3. "/url#something" for anchor on a page (current or not)
    # 4. "https://google.com#something" for external URL
    if menu_id.page_id:
        menu_id.page_id = None
    if menu['url'].startswith('#') and len(menu['url']) > 1:
        # Case 2.: prefix anchor with referer URL
        menu['url'] = request.referer + menu['url']
else:
    # current code 
```
Something like that (written here so not tested, and need to check how to read referer exactly)

Reply:
@rdeodoo I tried to `import pudb;pu.db` and check in the console if request.referer existed but it told me that request didn't exist.

And it made me think, since this is inside a model, can we really use request there? Couldn't there be a case were the request object doesn't exist (e.g. an automated action, or a test's setup?)?

Instead I used the context, and checked if the context key existed. I kept the referer name for the key but there might be a better suited name.

Let me know what you think!

Reply:
> it told me that request didn't exist.

Did you import it?

> Couldn't there be a case were the request object doesn't exist (e.g. an automated action, or a test's setup?)?

The save method is only meant to be called by our frontend, so there will always be a request bound.
So, in master we would not really care about that. But in 14.0 you should protect it with a check condition on `request` indeed. It will always exists but will be falsy when not bound.